src/components/organisms/NavigatorPane/KindRenderer/KindRenderer.tsx

```diff
@@ -0,0 +1,69 @@
+import {useCallback, useState} from 'react';
+
+import {useAppDispatch, useAppSelector} from '@redux/hooks';
+import {collapseResourceKinds, expandResourceKinds} from '@redux/reducers/ui';
+import {getResourceMetaFromState} from '@redux/selectors/resourceGetters';
+
+import * as S from './KindRenderer.styled';
+import KindSuffix from './KindSuffix';
+import ResourceCounter from './ResourceCounter';
+
+interface KindRendererProps {
+  kind: string;
+}
+
+function KindRenderer(props: KindRendererProps) {
+  const {kind} = props;
+  const dispatch = useAppDispatch();
+  const [isHovered, setIsHovered] = useState<boolean>(false);
+
+  const isSelected = useAppSelector(
+    state =>
+      state.main.selection?.type === 'resource' &&
+      getResourceMetaFromState(state, state.main.selection.resourceIdentifier)?.kind === kind
+  );
+  const isHighlighted = useAppSelector(state =>
+    state.main.highlights.some(
+      highlight =>
+        highlight.type === 'resource' && getResourceMetaFromState(state, highlight.resourceIdentifier)?.kind === kind
+    )
+  );
+  const isCollapsed = useAppSelector(state => state.ui.navigator.collapsedResourceKinds.includes(kind));
+
+  const toggleCollapse = useCallback(() => {
+    if (isCollapsed) {
```

Comment:
We could maybe use a ref here with the `useSelectorWithRef` ?

Reply:
`isCollapsed` is used for rendering, so we need it to be reactive